### ISSUE TITLE
LineEdit shows PopupMenu and Timer as child nodes. Causes issues when used as tool

### ISSUE THREAD

**Issue by wyattbiker** [godotengine/godot#34049]
**Godot version:**
3.2.beta1: 
**OS/device including version:**
mac os x mojave

**Issue description:**
When accessing `LineEdit` child nodes, both `PopupMenu` and `Timer` show up as child nodes, causes issues when used as `tool`

**Steps to reproduce:**
Create a `Control` scene and add `LineEdit` as the sole sub node.
Then add the following code in LineEdit.gd and run the scene.
```python
extends LineEdit

func _ready() -> void:
	print("ok ready tree entered a=",self.name)
	print('scene owner',get_tree().get_edited_scene_root())
	print('self owner=',self.owner)
	for child in self.get_children():
		print('child name=',child)
		pass
	return

```

The output will be as follows:

```
ok ready tree entered a=LineEdit
scene owner[Object:null]
self owner=[Control:1182]
child name=[Timer:1184]
child name=[PopupMenu:1185]
```

In addition if `LineEdit` were made into its own scene and the script was changed as below as a tool, then anytime that scene was added to `Control` scene it would also add the `PopupMenu` and `Timer`:

```python
tool
extends LineEdit

func _enter_tree():
	print("ok ready tree entered a=",self.name)
	print('scene owner',get_tree().get_edited_scene_root())
	print('self owner=',self.owner)
	self.owner=get_tree().get_edited_scene_root()	
	for child in self.get_children():
		print('child name=',child.name)
		# causes PopupMenu and Timer to be added to scene
		child.owner=get_tree().get_edited_scene_root()
		pass
	return

```

**Comment by Zylann** [comment]
`LineEdit` is not the only node having child "composite nodes", and is not the last to do this. It's quite common for nodes to create child nodes dynamically for their own inner workings, even in GDScript plugins. Built in nodes doing this include `FileDialog`, `ColorPicker`, `ScrollContainer`, `TextEdit`, `WindowDialog`, `GraphNode`, `ItemList`... the list goes on and on.

I'm not sure to understand what problem it causes here. They appear as children... so what?
They don't appear in the scene tree in the first place *precisely because their owner is not set to be the edited scene*. So if you do it... yeah, they will show up and end up as dupes. But I'm not sure what would be the reason to do that.

**Comment by KoBeWi** [comment]
This is a loose duplicate of #7142
There's a PR solving this problem that has yet to be merged #30391
